### ISSUE TITLE
build_runner fail when using local package inside pubspec.yaml

### ISSUE THREAD

**Issue by EArminjon** [dart-lang/pub#4016]
* Dart SDK Version 
  - 2.19.6
* What package(s) from this repo you are using, and the version : 
  - build_runner: ^2.2.1
* What builder(s) you are using (or writing yourself). 
  - Freezed
* Whether you are using Windows, MacOSX, or Linux (if applicable)
  - MacOs
* Any other packages or constraints we should know about
  - I use a local package (a fork) of a dependency from pub.dev

I cannot run the following command : 

```
flutter pub run build_runner build --delete-conflicting-output
```
```
The pubspec.lock file has changed since the .dart_tool/package_config.json file was generated, please run "flutter pub get" again.
pub finished with exit code 65
```

I got this error since a add a local package inside my pubspec.yaml
```yaml
dependencies:
  huawei_location:
    path: forked/huawei_location
```

In attachment the zip of the fork (it's light) : [huawei_location.zip](https://github.com/dart-lang/build/files/12599777/huawei_location.zip)

**Comment by jakemac53** [comment]
I went ahead and transferred this to the pub repo

**Comment by jakemac53** [comment]
> ```yaml
> flutter:
>   uses-material-design: true
>   generate: true
> ```

The `generate: true` could be part of the issue, this has historically been problematic as it would end up doing some nasty hacks with your package config.

**Comment by EArminjon** [comment]
I tried to remove the generate:true, despite multiple flutter pub get the issue remain :/

**Comment by jakemac53** [comment]
Can you `flutter pub run` any other tools? For instance `flutter pub run test`?

**Comment by EArminjon** [comment]
flutter test fail because of missing file (not generated because of previous issue)

**Comment by jakemac53** [comment]
Yeah, so that confirms this is not related to build_runner but pub in some way.

**Comment by EArminjon** [comment]
To bring more data : I well tested to copy & code the local repo on a github repo to use the 'git' instead of 'path' property and it well work. So yep, real issue around the path property.

**Comment by sigurdm** [comment]
Do you think you can narrow this down to a small example that we can use for reproduction?

**Comment by EArminjon** [comment]
Yep 
https://github.com/EArminjon/flutter-issue

**Comment by sigurdm** [comment]
Yeah, I think this is related to flutter_gen, see also (https://github.com/dart-lang/pub/issues/3314). I think you can use flutter_gen_runner instead.

fwiw I cannot reproduce with your example. What version of flutter are you using.

```
git clone https://github.com/EArminjon/flutter-issue
Cloning into 'flutter-issue'...
[...]
> cd flutter-issue/
> flutter pub get

┌─────────────────────────────────────────────────────────┐
│ A new version of Flutter is available!                  │
│                                                         │
│ To update to the latest version, run "flutter upgrade". │
└─────────────────────────────────────────────────────────┘
Resolving dependencies... (3.5s)
[...]
Changed 8 dependencies!
> flutter pub run build_runner build --delete-conflicting-outputs
[...]
[INFO] Succeeded after 126ms with 0 outputs (0 actions)
⏎                                                                                                                              
sigurdm@elevator ~/p/flutter-issue (master)> flutter --version
Flutter 3.10.5 • channel stable • https://github.com/flutter/flutter
Framework • revision 796c8ef792 (3 months ago) • 2023-06-13 15:51:02 -0700
Engine • revision 45f6e00911
Tools • Dart 3.0.5 • DevTools 2.23.1
```

**Comment by EArminjon** [comment]
I use flutter 3.7.12

**Comment by EArminjon** [comment]
Fixed in 3.10.5
